CMakeLists.txt

```diff
@@ -197,6 +198,15 @@ The listed MPI implementation does not provide the required mpi.mod interface. \
 When using the GNU compiler in combination with Intel MPI, please use the \
 Intel MPI compiler wrappers. Check the INSTALL.md for more information.")
   endif ()
+  if (USE_MPI_F08)
+    if (NOT MPI_Fortran_HAVE_F08_MODULE)
+      message(
+        WARNING
```

Comment:
Please make this a hard fail. I always find it irritating if a software does some kind of fallback contradicting my explicit intent.

Reply:
@fstein93 could you consider @dev-zero change in this PR? I'm happy to merge it afterwards...
For the tests, I will open a specific issue and rely on CP2K tests for the time being...

Reply:
I will take care of it at the end of this week or begin of next week as I am currently relocating.

Reply:
OK, I did the change in the #687 
Thanks @fstein93 !